sphinxext/opengraph/__init__.py

```diff
@@ -67,26 +68,17 @@ def get_tags(
     # type tag
     tags["og:type"] = config["ogp_type"]
 
-    if os.getenv("READTHEDOCS") and config["ogp_site_url"] is None:
-        # readthedocs uses html_baseurl for sphinx > 1.8
-        parse_result = urlparse(config["html_baseurl"])
-
+    # url tag
+    # Get the canonical URL if ogp_site_url is not configured
+    if config["ogp_site_url"] is None:
+        # we require that either ogp_site_url or html_baseurl be configured
         if config["html_baseurl"] is None:
-            raise EnvironmentError("ReadTheDocs did not provide a valid canonical URL!")
-
-        # Grab root url from canonical url
-        config["ogp_site_url"] = urlunparse(
-            (
-                parse_result.scheme,
-                parse_result.netloc,
-                parse_result.path,
-                "",
-                "",
-                "",
+            raise ConfigError(
+                "A URL has not been configured and is required as per the OpenGraph Specification!"
             )
```

Comment:
We should mention that it should be ignored if deployed on RTD environments. And it should be an INFO. If this is a warning or error, it should not warn or error in CI

Reply:
Not 100% sure how this should be, ideally it should fail builds and CI when it's omitted, but we can't do that because of RTD cases. I'll probably make it a warning.